### PR TITLE
Unused store eliminator.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libyul/optimiser/UnusedStoreEliminator.cpp [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+			return true;
+		// Condition (i = cover_i_ng, e = cover_e_d):
+		// i.start <= e.start && e.start + e.length <= i.start + i.length
+		if (!_covered.start || !_covering.start || !_covered.length || !_covering.length)
+			return false;
+		optional<u256> startDiff = knowledge.differenceIfKnownConstant(*_covered.start, *_covering.start);
+		optional<u256> coveredLength = knowledge.valueIfKnownConstant(*_covered.length);
+		optional<u256> lengthDiff = knowledge.differenceIfKnownConstant(*_covering.length, *_covered.length);
+		if (
+			(startDiff && coveredLength && lengthDiff) &&
```
Comment (chriseth) [inline libyul/optimiser/UnusedStoreEliminator.cpp:412]: We should cehck this, too.

Comment (cameel) [reply libyul/optimiser/UnusedStoreEliminator.cpp:412]: What about that Z3 proof you mentioned earlier on the chat?

Comment (chriseth) [reply libyul/optimiser/UnusedStoreEliminator.cpp:412]: Yes, that's what I mean.

**Comment by chriseth** [comment]
Going back to the original PR.

**Pull request closed without merging**